pkg/controller/clusters/management.go

```diff
@@ -0,0 +1,37 @@
+package clusters
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+)
+
+// FetchManagementEksaCluster returns the management cluster object for a given workload Cluster.
+func FetchManagementEksaCluster(ctx context.Context, cli client.Client, cluster *v1alpha1.Cluster) (*v1alpha1.Cluster, error) {
+	mgmtCluster := &v1alpha1.Cluster{}
+	mgmtClusterKey := client.ObjectKey{
+		Namespace: cluster.Namespace,
+		Name:      cluster.ManagedBy(),
+	}
+	err := cli.Get(ctx, mgmtClusterKey, mgmtCluster)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+	if apierrors.IsNotFound(err) {
+		notFoundErr := fmt.Errorf("unable to retrieve management cluster %v: %v", cluster.Spec.ManagementCluster.Name, err)
+		clusterList := &v1alpha1.ClusterList{}
+		if err = cli.List(ctx, clusterList,
+			client.MatchingFields{"metadata.name": cluster.Spec.ManagementCluster.Name}); err != nil || len(clusterList.Items) == 0 {
```

Comment:
I think if err != nil we should return the error itself, not the precomputed notFound error

Reply:
Well this is a best effort case, so I decided that we take the error from the Get call instead of the error from the List, which is why I decided the save the error from before. However, we can return the error from list instead if we want to prioritize that error over not finding it from the same namespace.